nsxt/resource_nsxt_policy_role.go

```diff
@@ -0,0 +1,266 @@
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/aaa"
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var featurePermissionTypes = [](string){
+	nsxModel.FeaturePermission_PERMISSION_CRUD,
+	nsxModel.FeaturePermission_PERMISSION_READ,
+	nsxModel.FeaturePermission_PERMISSION_EXECUTE,
+}
+
+func resourceNsxtPolicyUserManagementRole() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyUserManagementRoleCreate,
+		Read:   resourceNsxtPolicyUserManagementRoleRead,
+		Update: resourceNsxtPolicyUserManagementRoleUpdate,
+		Delete: resourceNsxtPolicyUserManagementRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"display_name": getDataSourceDisplayNameSchema(),
+			"description":  getDataSourceDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"role": {
+				Type:        schema.TypeString,
+				Description: "Short identifier for the role. Must be all lower case with no spaces",
+				Required:    true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(
+						"^[_a-z0-9-]+$"),
+					"Must be a valid role identifier matching: ^[_a-z0-9-]+$"),
+			},
+			"feature": {
+				Type:        schema.TypeList,
+				Description: "List of permissions for features",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"feature": {
+							Type:        schema.TypeString,
+							Description: "Feature Id",
+							Required:    true,
+						},
+						"feature_description": {
+							Type:        schema.TypeString,
+							Description: "Feature Description",
+							Computed:    true,
+						},
+						"feature_name": {
+							Type:        schema.TypeString,
+							Description: "Feature Name",
+							Computed:    true,
+						},
+						"permission": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(featurePermissionTypes, false),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceNsxtPolicyUserManagementRoleExists(id string, connector client.Connector) (bool, error) {
+	var err error
+	roleClient := aaa.NewRolesClient(connector)
+	_, err = roleClient.Get(id)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func getFeaturePermissionFromSchema(d *schema.ResourceData) []nsxModel.FeaturePermission {
+	features := d.Get("feature").([]interface{})
+	featurePermissions := make([]nsxModel.FeaturePermission, 0, len(features))
+	for _, feature := range features {
+		data := feature.(map[string]interface{})
+		featureID := data["feature"].(string)
+		featureDescription := data["feature_description"].(string)
```

Comment:
since `feature_description` and `feature_name` are Computed-only, we don't need those here I think

Reply:
done